File: tests/Contexts/Rutine/Exercises/domain/ExerciseMother.ts
import { Exercise } from '../../../../../src/Contexts/Rutine/Exercises/domain/Exercise';
import { ExerciseBar } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseBar';
import { ExerciseId } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseId';
import { ExerciseMuscle } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseMuscle';
import { ExerciseName } from '../../../../../src/Contexts/Rutine/Exercises/domain/ExerciseName';
import { ExerciseBarMother } from './ExerciseBarMother';
import { ExerciseIdMother } from './ExerciseIdMother';
import { ExerciseMuscleMother } from './ExerciseMuscleMother';
import { ExerciseNameMother } from './ExerciseNameMother';

export class ExerciseMother {
  static create(id: ExerciseId, name: ExerciseName, muscle: ExerciseMuscle, bar: ExerciseBar) {
    return new Exercise(id, name, muscle, bar);
  }

  static random(): Exercise {
    return this.create(
      ExerciseIdMother.random(),
      ExerciseNameMother.random(),
      ExerciseMuscleMother.random(),
      ExerciseBarMother.random()
    );
  }
}

File: src/test/java/com/kuro/rutineweb/exercises/entities/ExerciseMother.java
package com.kuro.rutineweb.exercises.entities;

import com.kuro.rutineweb.Shared.entities.UserId;

public class ExerciseMother {
    public static Exercise create(UserId userId, ExerciseId id, ExerciseName name, ExerciseMuscle muscle, ExerciseBar bar, ExerciseRm rm) {
        return new Exercise(userId, id, name, muscle, bar, rm);
    }

    public static Exercise random() {
        return create(
                UserIdMother.random(),
                ExerciseIdMother.random(),
                ExerciseNameMother.random(),
                ExerciseMuscleMother.random(),
                ExerciseBarMother.random(),
                ExerciseRmMother.random()
        );
    }
}
